stl/inc/memory

```diff
@@ -3154,7 +3154,7 @@ protected:
         }
     }
 
-    _Ty* _Ptr = nullptr;
+    atomic<_Ty*> _Ptr{nullptr};

```

Comment:
However, I wonder whether it should be named something like `_Atom` now? (This is because the code below has to access both it and `shared_ptr`'s `_Ptr` member, but I think of `atomic` as being extra special and using the exact same data member name seems confusing.) I defer to your and @BillyONeal's opinions here.

Reply:
`atomic` is here not used for ordering or for CAS/RMW, it is just to prevent hypothetical tearing.

The tearing is also prohibited by WIndows Docs (see [Interlocked Variable Access
](https://docs.microsoft.com/en-us/windows/win32/sync/interlocked-variable-access):
> Simple reads and writes to properly-aligned 32-bit variables are atomic operations. 
> ...
> Simple reads and writes to properly aligned 64-bit variables are atomic on 64-bit Windows. 

Yet to avoid relying on that, I prefer explicit `atomic`.

It does make sense to rename, to make sure that only `load` and `store` are used. Implicit `operator T` and `operator=` are sequentially consistent, which is superfluous here.

Reply:
I mildly prefer this as submitted by Alex; naming it `_Atom` seems kinda like hungarian notation to me. This is always the same as the pointer member, we just happen to access it atomically. _Rep is substantially weirder since we're stealing its low order bit so I could see wanting to rename that but we kinda already gave it a slightly different `_Repptr` name.

Reply:
Ok, thanks for explaining.